zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/batchoperation/BatchOperationExecuteProcessor.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.batchoperation;
+
+import io.camunda.zeebe.engine.processing.ExcludeAuthorizationCheck;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedCommandWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.batchoperation.PersistedBatchOperation;
+import io.camunda.zeebe.engine.state.immutable.BatchOperationState;
+import io.camunda.zeebe.engine.state.immutable.ProcessingState;
+import io.camunda.zeebe.protocol.impl.record.value.batchoperation.BatchOperationExecutionRecord;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceRecord;
+import io.camunda.zeebe.protocol.record.intent.BatchOperationExecutionIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ExcludeAuthorizationCheck
+public final class BatchOperationExecuteProcessor
+    implements TypedRecordProcessor<BatchOperationExecutionRecord> {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(BatchOperationExecuteProcessor.class);
+
+  private static final int BATCH_SIZE = 10;
+
+  private final TypedCommandWriter commandWriter;
+  private final StateWriter stateWriter;
+  private final int partitionId;
+  private final BatchOperationState batchOperationState;
+
+  public BatchOperationExecuteProcessor(
+      final Writers writers, final ProcessingState processingState, final int partitionId) {
+    commandWriter = writers.command();
+    stateWriter = writers.state();
+    batchOperationState = processingState.getBatchOperationState();
+    this.partitionId = partitionId;
+  }
+
+  @Override
+  public void processRecord(final TypedRecord<BatchOperationExecutionRecord> command) {
+    final var recordValue = command.getValue();
+    LOGGER.debug(
+        "Processing new command with key '{}' on partition{} : {}",
+        command.getKey(),
+        partitionId,
+        recordValue);
+    final long batchKey = command.getValue().getBatchOperationKey();
```

Comment:
```suggestion
    final long batchKey = recordValue.getBatchOperationKey();
```
🔧 you already extracted the value, better re-use it below